lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -687,9 +687,23 @@ def generate_clone_spec(src_config)
         # We should get here with the customizations set, either by a plugin or a --cspec
         fatal_exit 'Windows clones need a customization identity. Try passing a --cspec or making a --cplugin' if cust_spec.identity.props.empty?
 
-        identification = RbVmomi::VIM.CustomizationIdentification(
-          joinWorkgroup: cust_spec.identity.identification.joinWorkgroup
-        )
+        # If --cdomain matches what is in --cspec then use identification from the --cspec, else use --cdomain
+        domain = get_config(:customization_domain)
+        if domain
+          if domain == cust_spec.identity.identification.joinDomain
```

Comment:
Solving that feedback doesn't feel very clean, but if an approach like this meets style checks, let me know and I'll change it out:

```
# If --cdomain matches what is in --cspec then use identification from the --cspec, else use --cdomain
identification = case domain = get_config(:customization_domain)
when nil?
  # Fall back to original behavior of using joinWorkgroup from the --cspec
  RbVmomi::VIM.CustomizationIdentification(
    joinWorkgroup: cust_spec.identity.identification.joinWorkgroup
  )
when cust_spec.identity.identification.joinDomain
  cust_spec.identity.identification
else
  RbVmomi::VIM.CustomizationIdentification(
    joinDomain: domain
  )
end
```